gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-scim/src/main/java/io/gravitee/am/gateway/handler/scim/service/impl/BulkServiceImpl.java

```diff
@@ -68,56 +70,14 @@ public class BulkServiceImpl implements BulkService {
-                                                    validOperation.setLocation(scimUser.getMeta().getLocation());
-                                                    validOperation.setStatus(valueOf(HttpStatusCode.OK_200));
-                                                    // response attribute is not set for successful operation
-                                                    // so no need to provide the scimUser in the operation.response
-                                                    return validOperation.asResponse();
-                                                });
-                                    case PATCH:
-                                        return new PatchUserAction(userService, domain, client)
-                                                .execute(extractUserIdFromPath(validOperation), updateBaseUrl(baseUrl, validOperation.getPath()), operation.getData(), authenticationContext, principal)
-                                                .map(scimUser -> {
-                                                    validOperation.setLocation(scimUser.getMeta().getLocation());
-                                                    validOperation.setStatus(valueOf(HttpStatusCode.OK_200));
-                                                    // response attribute is not set for successful operation
-                                                    // so no need to provide the scimUser in the operation.response
-                                                    return validOperation.asResponse();
-                                                });
-                                    case DELETE:
-                                        return userService.delete(extractUserIdFromPath(validOperation), principal)
-                                                .andThen(Single.fromSupplier(() -> {
-                                                    validOperation.setStatus(valueOf(HttpStatusCode.NO_CONTENT_204));
-                                                    validOperation.setLocation(updateBaseUrl(baseUrl, validOperation.getPath()));
-                                                    return validOperation.asResponse();
-                                                }));
-                                    default:
-                                        io.gravitee.am.gateway.handler.scim.model.Error error = new Error();
-                                        error.setScimType("invalidSyntax"); // should not happen
-                                        validOperation.setResponse(error);
-                                        return Single.just(validOperation.asResponse());
-
-                                }
-                            })
-                            .onErrorResumeNext(ex-> {
+                .concatMapMaybe(operation -> Single.just(operation)
+                            .filter(x -> failOnErrors == null || failOnErrors <= 0 || failuresCounter.get() < failOnErrors)
```

Comment:
should we allow negative values for `failOnErrors`?
I think it would be better to change the `BulkRequest#getFailOnErrors()` to always return a non-negative int (for null, we could default to `Integer.MAX_VALUE`)